gemfire-shared/src/main/java/com/gemstone/gemfire/internal/shared/LauncherBase.java

```diff
@@ -387,14 +387,15 @@ protected long getLastModifiedStatusNanos() throws IOException {
    * Verify and clear the status. If a server is detected as already running
    * then returns an error string else null.
    */
-  protected String verifyAndClearStatus() throws IOException {
+  protected int verifyAndClearStatus() throws IOException {
     final Status status = getStatus();
     if (status != null && status.state != Status.SHUTDOWN) {
-      return MessageFormat.format(LAUNCHER_IS_ALREADY_RUNNING_IN_DIRECTORY,
-          this.baseName, getWorkingDirPath());
+      System.err.println(MessageFormat.format(LAUNCHER_IS_ALREADY_RUNNING_IN_DIRECTORY,
+          this.baseName, getWorkingDirPath(), status.getStateString(status.state)));
+      return status.state;
     }
     deleteStatus();
-    return null;
+    return status.state;
```

Comment:
Since there is a null check above, this can cause NPE. Better to always return Status.SHUTDOWN directly.